src/Compiler/Graphs/MinCut.cs

```diff
@@ -198,10 +199,8 @@ public Set<NodeType> GetSourceGroup()
                     foreach (EdgeType edge in graph.EdgesInto(node))
                     {
                         NodeType target = graph.SourceOf(edge);
-                        if (distanceToSink[node] != distanceToSink[target] + 1) continue;
-                        // target cannot be in sourceGroup since their distances are huge,
-                        // unless it is both a source and sink.
-                        if (Sources.Contains(target)) continue;
+                        if (sourceGroup.Contains(target)) continue;
```

Comment:
Now that the change is simpler, I think the original code and comment should be restored.